content/en/docs/Core concepts/metrics.md

```diff
@@ -0,0 +1,125 @@
+---
+title: "Metrics"
+date: 2020-11-03
+weight: 2
+description: >
+  How to export metrics from Baker
+---
+
+### Metrics
+
+ During execution, Baker collects different kind of performance data points:
+
+ - general pipeline metrics such as the total number of records processed, records per seconds.
+ - component-specific metrics: files written per second, discarded records (in a filter), errors, etc.
+ - Go runtime metrics, mallocs, frees, garbage collections and so on.
+
+If enabled, Baker collects all these metrics and forwards them to a metrics client.
+
+When configured, a Baker topology also configures the metrics client to use, if any.
+Metrics configuration is performed in Baker configuration TOML file.
+
+
+#### How to export metrics?
+
+When running a Baker pipeline exports metrics about the Go runtime as well as general 
+metrics giving an high-level overview of the ongoing job.
+
+More specific metrics are also exported on a per-component basis. To that effect, 
+`baker.Input`, `baker.Filter`, `baker.Output` and `baker.Upload` all have a `Stats` 
+method. `Stats` is called every second and the component is expected to return both
+a predefined set of metrics and a [`baker.MetricsBag`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsBag)
+containing metrics of arbitrary name and types.
+
+Let's illustrate this with metrics exported by a filter via 
+[`baker.FilterStats`](https://pkg.go.dev/github.com/AdRoll/baker#FilterStats):
+
+```go
+type FilterStats struct {
+	NumProcessedLines int64
+	NumFilteredLines  int64
+	Metrics           MetricsBag
+}
+```
+
+In this case `NumProcessedLines` must represent the **total** number of processed 
+lines since Baker started, and `NumFilteredLines` is the number of discarded 
+(or filtered) records. Due to hstorical reasons these fields have lines in them
+but they do mean the number of records.
+
+An important point is that `Stats` can be called from any goroutine so it must be
+safe for concurrent use by multiple goroutines. 
+
+```go
+func (i *myFilter) Stats() baker.FilterStats {
+    bag := make(baker.MetricsBag)
+    bag.AddGauge("current_speed", float64(atomic.LoadInt64(&myFilter.speed)))
+
+    return baker.FilterStats{
+        NumProcessedLines: atomic.LoadInt64(&myFilter.totalLines),
+        NumFilteredLines: atomic.LoadInt64(&myFilter.filteredLines),
+        // Metrics could be let to its default value, nil, if not needed.
+        Metrics: bag,
+    }
+}
+```
+
+#### Configuring metrics in TOML
+
+Baker configuration TOML files may have a `[metrics]` section dedicated to the 
+configuration of a metrics client.
+
+`[metrics.name]` specifies the metrics client to use, from the list of all registered [`baker.MetricsClient`](https://pkg.go.dev/github.com/AdRoll/baker#MetricsClient).
```

Comment:
if you want to force a new line at the end, you should add double space or the sentences will be in the same line